media-gfx/yafaray/yafaray-3.3.0.ebuild

```diff
@@ -0,0 +1,103 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{5,6} )
+inherit cmake-utils python-r1 python-utils-r1
+
+DESCRIPTION="A free open-source montecarlo raytracing engine"
+HOMEPAGE="http://www.yafaray.org"
+SRC_URI="https://github.com/YafaRay/Core/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/YafaRay/Blender-Exporter/archive/v${PV}.tar.gz -> ${PN}-blender-exporter-${PV}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+blender debug +fastmath +fasttrig jpeg opencv openexr png +python -ruby tiff truetype"
+
+RDEPEND="dev-libs/boost
+	dev-libs/libxml2
+	sys-libs/zlib
+	blender? ( media-gfx/blender )
+	jpeg? ( virtual/jpeg:0 )
+	opencv? ( >=media-libs/opencv-3.1.0[openexr?] )
+	openexr? ( >=media-libs/openexr-2.2.0 )
+	png? ( media-libs/libpng:= )
+	python? ( ${PYTHON_DEPS} )
+	ruby? ( dev-lang/ruby:= )
+	tiff? ( media-libs/tiff:0 )
+	truetype? ( media-libs/freetype )"
+DEPEND="${RDEPEND}
+	dev-lang/swig"
+
+S="${WORKDIR}/Core-${PV}"
+
+src_prepare() {
+	cmake-utils_src_prepare
+
+	sed -i "s/-Wall/${CXXFLAGS}/g" CMakeLists.txt || die # respect flags
+	sed -i "s/@YAFARAY_BLENDER_EXPORTER_VERSION@/v${PV}/" "${WORKDIR}/Blender-Exporter-${PV}/__init__.py" || die
+
+	pushd "${WORKDIR}/Blender-Exporter-${PV}"
+	eapply "${FILESDIR}/${P}-blender-exporter-paths.patch"
+	popd
+
+	eapply "${FILESDIR}/${P}-ruby-includes.patch"
+
+	eapply_user
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DYAF_LIB_DIR=$(get_libdir)
+		-DYAF_BINDINGS_RUBY_DIR="$(ruby -e 'print RbConfig::CONFIG["vendorarchdir"]')"
```

Comment:
You don't want to do that with `USE=-ruby`. Also, you don't really use the version you depend, I think.

Reply:
I was very uncertain about this part. I really don't use Ruby often enough to know best practice here. How should I conditionally add to mycmakeargs? What's the best way to get the correct directory to install to?

I'm going to ask around and see what I can find out.

Reply:
I don't know much about Ruby either, and I think that's one of the cases that aren't really correctly covered in Gentoo. So the easy answer is: disable Ruby bindings until somebody really needs them.

Reply:
Sounds good.